src/syscalls.rs

```diff
@@ -43,7 +44,10 @@ pub fn yieldk_for<F: Fn() -> bool>(cond: F) {
     }
 }
 
-pub fn subscribe<CB: SubscribableCallback>(mut callback: CB) -> (isize, CallbackSubscription<CB>) {
+pub fn subscribe<I: SubscribeInfo, CB: SubscribableCallback>(
+    subscribe_info: I,
+    callback: &mut CB,
+) -> Result<CallbackSubscription<I>, isize> {
     extern "C" fn c_callback<CB: SubscribableCallback>(
```

Comment:
I think when we pass a pointer to c_callback to the kernel it should be something statically valid. I would make it a function in the module.

Reply:
It _is_ statically valid but to make this clear I can put the function outside.